src/App.tsx

```diff
@@ -5,28 +5,67 @@ import { parseOpenRPCDocument } from '@open-rpc/schema-utils-js';
 import React, { useEffect, useState } from 'react';
 
 import './App.css';
+import type { JsonValue } from './constants/signingMethods';
+import {
+  insertSigningAddress,
+  SIGNING_METHODS,
+} from './constants/signingMethods';
 import { openRPCExampleToJSON } from './helpers/OpenRPCExampleToJSON';
 import MetaMaskMultichainProvider from './providers/MetaMaskMultichainProvider';
 import makeProvider from './providers/MockMultichainProvider';
 import type { Provider } from './providers/Provider';
 
+const truncateJSON = (
+  json: any,
+  maxLength = 100,
+): { text: string; truncated: boolean } => {
+  const stringified = JSON.stringify(json, null, 2);
+  if (stringified.length <= maxLength) {
+    return { text: stringified, truncated: false };
+  }
+  return {
+    text: stringified,
+    truncated: true,
+  };
+};
+
+const FEATURED_NETWORKS = {
+  'eip155:1': 'Ethereum Mainnet',
+  'eip155:59144': 'Linea Mainnet',
+  'eip155:42161': 'Arbitrum One',
+  'eip155:43114': 'Avalanche Network C-Chain',
+  'eip155:56': 'BNB Chain',
+  'eip155:10': 'OP Mainnet',
+  'eip155:137': 'Polygon Mainnet',
+  'eip155:324': 'zkSync Era Mainnet',
+  'eip155:8453': 'Base Mainnet',
+  'eip155:1337': 'Localhost',
+} as const;
+
 function App() {
   const [createSessionResult, setCreateSessionResult] = useState<any>(null);
   const [providerType, setProviderType] = useState<string>('metamask');
   const [provider, setProvider] = useState<Provider>();
   const [getSessionResult, setGetSessionResult] = useState<any>(null);
-  const [revokeSessionResult, setRevokeSessionResult] = useState<any>(null);
+  // const [revokeSessionResult, setRevokeSessionResult] = useState<any>(null);
   const [selectedMethods, setSelectedMethods] = useState<
     Record<string, string>
   >({});
   const [invokeMethodResults, setInvokeMethodResults] = useState<
-    Record<string, any>
+    Record<string, Record<string, any[]>>
   >({});
   const [customScope, setCustomScope] = useState<string>('');
   const [selectedScopes, setSelectedScopes] = useState<Record<string, boolean>>(
     {
-      'eip155:1337': false,
       'eip155:1': false,
+      'eip155:59144': false,
+      'eip155:42161': false,
+      'eip155:43114': false,
+      'eip155:56': false,
+      'eip155:10': false,
+      'eip155:137': false,
+      'eip155:324': false,
+      'eip155:8453': false,
```

Comment:
does `eip155:1337` need to be in here?

Reply:
Yes for e2e testing!

Reply:
Well I think we're going to have to think through whether we actually have ganache emulate other chainIds in e2e testing

Reply:
eh it can be 1337, 1338, ... etc

Reply:
oh wait... these chainIds need to be hex

Reply:
Why? These are the CAIP-2 ideas that are used in the CAIP-25 request?

Reply:
Nevermind. I'm being dumb